app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.html

```diff
@@ -59,63 +59,144 @@
-               required
-               class="form-control input-sm"
-               name="gitRepo"
-               ng-model="command.repositoryUrl"/></div>
+        <div class="col-md-1">
+          <input type="radio"
+                 ng-model="command.sourceType"
+                 value="gcs">
+        </div>
+        <div class="col-md-2">GCS</div>
+
+        <div class="col-md-1">
+          <input type="radio"
+                 ng-model="command.sourceType"
+                 value="git">
+        </div>
+        <div class="col-md-2">Git</div>
+      </div>
     </div>
 
-    <div class="form-group">
-      <div class="col-md-3 sm-label-right">
-        Git Credential Type <help-field class="help-field-absolute" key="appengine.serverGroup.gitCredentialType"></help-field>
+    <div ng-if="basicSettingsCtrl.isGcsSource()">
+      <div class="form-group row">
+        <label class="col-md-3 sm-label-right">Storage Account</label>
+        <div class="col-md-7">
+          <select class="form-control input-sm"
+                  ng-options="option for option in command.backingData.storageAccounts"
+                  ng-model="command.storageAccountName">
+            <option value="">Select...</option>
+          </select>
+        </div>
       </div>
-      <div class="col-md-7">
-        <select class="form-control input-sm"
-                ng-options="basicSettingsCtrl.humanReadableGitCredentialType(type) for type in basicSettingsCtrl.getSupportedGitCredentialTypes()"
-                ng-model="command.gitCredentialType">
-        </select>
+
+      <div class="form-group row">
+        <label class="col-md-3 sm-label-right">Resolve URL</label>
+        <div class="col-md-7">
+          <div class="radio">
+            <label>
+              <input type="radio"
+                     ng-model="command.fromArtifact"
+                     ng-value="false"> via text input
+            </label>
+          </div>
+          <div class="radio">
+            <label>
+              <input type="radio"
+                     ng-model="command.fromArtifact"
+                     ng-value="true"> via pipeline artifact
+            </label>
+          </div>
+        </div>
+      </div>
+
+      <div class="form-group" ng-if="command.fromArtifact">
+        <label class=" col-md-3 sm-label-right">Expected Artifact</label>
+        <div class="col-md-7">
+          <ui-select ng-model="command.expectedArtifact"
+                     class="form-control input-sm">
+            <ui-select-match>{{ basicSettingsCtrl.summarizeExpectedArtifact(command.expectedArtifact) }}
+            </ui-select-match>
+            <ui-select-choices repeat="expected in command.backingData.expectedArtifacts">
+              <span>{{ basicSettingsCtrl.summarizeExpectedArtifact(expected) }}</span>
+            </ui-select-choices>
+          </ui-select>
+        </div>
+      </div>
+
+      <div class="form-group" ng-if="!!!command.fromArtifact">
```

Comment:
Not not not?

Reply:
I think I tried to use `!!` to cast this at some point and forgot to change it back.